src/simularium/VisDataCache.ts

```diff
@@ -148,36 +130,40 @@ class VisDataCache {
     }
 
     private getFrameAtCondition(
-        condition: (data: CacheNode) => boolean,
-        value: number
-    ): CachedFrame {
+        condition: (data: CacheNode) => boolean
+    ): CachedFrame | undefined {
         if (!this.head) {
-            return this.frameAccessError("No data in cache.");
+            return;
         }
-        const frame = this.walkLinkedList(condition);
+        const frame = this.findInLinkedList(condition);
         if (frame) {
             return frame.data;
         }
-        return this.frameAccessError(
-            `Frame not found at provided ${condition}. Attempting to access frame ${value}.`
-        );
     }
 
-    public getFrameAtTime(time: number): CachedFrame {
-        return this.getFrameAtCondition(
-            (node) => compareTimes(node.data.time, time, 0) === 0,
-            time
+    public getFrameAtTime(time: number): CachedFrame | undefined {
+        // return this.getFrameAtCondition(
+        //     (node) => compareTimes(node.data.time, time, 0) === 0,
+        //     time
+        // );
+        const frame = this.getFrameAtCondition(
+            (node) => compareTimes(node.data.time, time, 0) === 0
         );
+        return frame ? frame : undefined;
     }
 
-    public getFrameAtFrameNumber(frameNumber: number): CachedFrame {
-        return this.getFrameAtCondition(
-            (node) => node.data["frameNumber"] === frameNumber,
-            frameNumber
+    public getFrameAtFrameNumber(frameNumber: number): CachedFrame | undefined {
+        // return this.getFrameAtCondition(
+        //     (node) => node.data["frameNumber"] === frameNumber,
+        //     frameNumber
+        // );
+        const frame = this.getFrameAtCondition(
+            (node) => node.data["frameNumber"] === frameNumber
         );
+        return frame ? frame : undefined;
     }
```

Comment:
Does the commented-out code want to stay?

Reply:
:')